projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/execution/JobExecutionResultManagerTest.java

```diff
@@ -0,0 +1,161 @@
+      Assertions.assertEquals(ExecutionTerminationStatus.NONE, actualTerminationStatus);
+   }
+
+   @Test
+   public void testGetTerminationStatus_terminationMessageSuccess_shouldReturnTerminationStatusSuccess() {
+      ExecutionTerminationStatus actualTerminationStatus = JobExecutionResultManager.getTerminationStatus(ExecutionTerminationStatus.SUCCESS.getString());
+      Assertions.assertEquals(ExecutionTerminationStatus.SUCCESS, actualTerminationStatus);
+   }
+
+   @Test
+   public void testUpdateTerminationStatusBasedOnExecutionStatus_emptyTerminationMessageAndExecutionStatusFinished_shouldReturnTerminationStatusSuccess() {
+      ExecutionTerminationStatus actualTerminationStatus =
+            JobExecutionResultManager.updateTerminationStatusBasedOnExecutionStatus(ExecutionTerminationStatus.NONE, ExecutionStatus.FINISHED, "");
+
+      Assertions.assertEquals(ExecutionTerminationStatus.SUCCESS, actualTerminationStatus);
+   }
+
+   @Test
+   public void testUpdateTerminationStatusBasedOnExecutionStatus_emptyTerminationMessageAndExecutionStatusFailed_shouldReturnTerminationStatusPlatformError() {
+      ExecutionTerminationStatus actualTerminationStatus =
+            JobExecutionResultManager.updateTerminationStatusBasedOnExecutionStatus(ExecutionTerminationStatus.NONE, ExecutionStatus.FAILED, "");
+
+      Assertions.assertEquals(ExecutionTerminationStatus.PLATFORM_ERROR, actualTerminationStatus);
+   }
+
+   @Test
+   public void testUpdateTerminationStatusBasedOnExecutionStatus_emptyTerminationMessageAndExecutionStatusRunning_shouldReturnTerminationStatusSuccess() {
+      ExecutionTerminationStatus actualTerminationStatus =
+            JobExecutionResultManager.updateTerminationStatusBasedOnExecutionStatus(ExecutionTerminationStatus.SUCCESS, ExecutionStatus.RUNNING, "");
+
+      Assertions.assertEquals(ExecutionTerminationStatus.SUCCESS, actualTerminationStatus);
+   }
+
+   @Test
+   public void testUpdateTerminationStatusBasedOnExecutionStatus_terminationMessageSuccessAndExecutionStatusFinished_shouldReturnTerminationStatusUserError() {
+      ExecutionTerminationStatus actualTerminationStatus =
+            JobExecutionResultManager.updateTerminationStatusBasedOnExecutionStatus(ExecutionTerminationStatus.USER_ERROR, ExecutionStatus.FINISHED, ExecutionTerminationStatus.SUCCESS.getString());
+
+      Assertions.assertEquals(ExecutionTerminationStatus.USER_ERROR, actualTerminationStatus);
+   }
+
+   @Test
+   public void testGetExecutionStatus_executionSucceededNull_shouldReturnExecutionStatusRunning() {
+      ExecutionStatus actualExecutionStatus = JobExecutionResultManager.getExecutionStatus(null);
+
+      Assertions.assertEquals(ExecutionStatus.RUNNING, actualExecutionStatus);
+   }
+
+   @Test
+   public void testGetExecutionStatus_executionSucceededTrue_shouldReturnExecutionStatusFinished() {
+      ExecutionStatus actualExecutionStatus = JobExecutionResultManager.getExecutionStatus(true);
```

Comment:
I find using private or package private methods in unit tests, make them more brittle and more prone to break when we refactor things. It's best if we test only the public interface (the public methods) of a class based on its expected behavior and effects. This way we can change the implementation and if we don't break anything the test would pass.

Reply:
I agree, but the logic here is quite confusing and I prefer to leave it as it is.

Reply:
Fixed